src/hooks/routing/clientSideSmartOrderRouter.ts

```diff
@@ -25,9 +26,11 @@ async function getQuote(
     tokenOut: { address: string; chainId: number; decimals: number; symbol?: string }
     amount: BigintIsh
   },
-  routerParams: AlphaRouterParams,
+  providerUrl: string,
   routerConfig: Partial<AlphaRouterConfig>
 ): Promise<{ data: GetQuoteResult; error?: unknown }> {
+  const provider = new JsonRpcProvider(providerUrl)
```

Comment:
we unfortunately can't -- provider (type BaseProvider from ethers) is non-serializable & redux store doesn’t like that if we try to pass a non-serializable obj into queryFn

cc @justindomingue

Reply:
We can't use the JsonRpcProvider from our web3 connectin?

Reply:
doesn't this run outside of the react lifecycle?

Reply:
it does, but we don't get rid of these providers - they're just garbage-collected.
I think it will be compute-intensive (and therefore maybe laggy) if we re-instantiate them every time.

Alternatively, can we have a file-level cache where we re-use a provider until the providerUrl changes?

Reply:
redux maintainer says about re-instantiating object every time queryFn is run — “it’s cached so you wouldn’t actually do that unless you’re making a network request. which only happens when you don’t have a cached value (i.e. nothing’s already subscribed to it)”